### PR TITLE
feat: Add Outlook Trigger Node

### PR DESCRIPTION
**Pull request by bramkn**
## Summary
Adds the Outlook Trigger Node

## Related tickets and issues
Linear: https://linear.app/n8n/issue/NODE-1109/outlook-trigger-node-new-node

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/nodes-base/nodes/Microsoft/Outlook/trigger/MessageDescription.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+								name: 'hasAttachments',
+								type: 'boolean',
+								default: false,
+							},
+							{
+								displayName: 'Folders to Exclude',
+								name: 'foldersToExclude',
+								type: 'multiOptions',
+								typeOptions: {
+									loadOptionsMethod: 'getFolders',
```
Comment (Joffcom) [inline packages/nodes-base/nodes/Microsoft/Outlook/trigger/MessageDescription.ts:75]: This isn't defined in the `loadOptions` method so currently doesn't work.

**Review by Joffcom — CHANGES_REQUESTED** [review]

File: packages/nodes-base/nodes/Microsoft/Outlook/trigger/MessageDescription.ts (hunk 1 above)
Comment (bramkn) [reply packages/nodes-base/nodes/Microsoft/Outlook/trigger/MessageDescription.ts]: Fixed, sorry for the inconvenience

**Review by Joffcom — APPROVED** [review]
Looks good @bramkn I resolved the merge conflicts and everything is working 🙌🏻

**Comment by janober** [comment]
Got released with [`n8n@1.31.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.31.0)

**Pull request merged**